Libraries/CLI/src/cli.c

```diff
@@ -0,0 +1,265 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL MAXIM INTEGRATED BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * Except as contained in this notice, the name of Maxim Integrated
+ * Products, Inc. shall not be used except as stated in the Maxim Integrated
+ * Products, Inc. Branding Policy.
+ *
+ * The mere transfer of this software does not imply any licenses
+ * of trade secrets, proprietary technology, copyrights, patents,
+ * trademarks, maskwork rights, or any other form of intellectual
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ ******************************************************************************/
+
+/* -------------------------------------------------- */
+//                      INCLUDES
+/* -------------------------------------------------- */
+
+#include "cli.h"
+
+/* -------------------------------------------------- */
+//             FUNCTION PROTOTYPES
+/* -------------------------------------------------- */
+
+/** Checks if there is white space present in the given character pointer
+ *
+ * @param p
+ * 		  character pointer containing the line accumulator input
+ *
+ * @return 1 if white space is present
+ * 		   0 if no white space is present
+ */
+bool white_space_present(char *p);
+
+/** Checks if there is no white space present in the given character pointer
+ *
+ * @param p
+ * 		  character pointer containing the line accumulator input
+ *
+ * @return 1 if no white space is present
+ * 		   0 if white space is present
+ */
+bool white_space_not_present(char *p);
+
+void User_Prompt_Sequence(void);
+
+void Console_Backspace_Sequence(void);
+
+void Console_Cmd_Clear(void);
+
+void Clear_buffer(void);
+
+/* -------------------------------------------------- */
+//                      GLOBALS
+/* -------------------------------------------------- */
+
+/*! \name buf
+    \brief Buffer to store the characters of the command 
+*/
+char buf[256];
+int idx = 0;
```

Comment:
One small suggestion here, we can use either uint16_t or uint32_t type for idx, since the array index cannot be negative.